src/mx/NotificationManager.java

```diff
@@ -0,0 +1,24 @@
+package mx;
+
+import java.util.ArrayList;
+
+/**
+ * Created by masterix on 28.05.2017.
+ */
+public class NotificationManager {
```

Comment:
Actually, this is not a manager - it's more of a repository, it's missing logic. I can imagine object of this class as one that maintains lists of e.g. `NotificationHandler`s, propagating events to them.
`NotificationHandler` would hold some `Predicate<DomainStatus>` and `NotificationSender`. Handler would be notified about status of a domain and if the predicate is fullfiled, it would use sender to send a notification. Of course `NotificationHandler` should be immutable. Handler should also encapsulate logic regarding transforming domain status to notification message.